### PR TITLE
1.3 small fixes for compatibility and performance

### PR DESCRIPTION
**Pull request by michaelneale**
# Description

Cherry pick of fixes for unit tests for newer core, a flaky test that PCT picked up, and log performance improvements. 

https://github.com/jenkinsci/blueocean-plugin/pull/1521
https://github.com/jenkinsci/blueocean-plugin/pull/1552
https://github.com/jenkinsci/blueocean-plugin/pull/1571

### CODE REVIEW

**Comment by michaelneale** [comment]
a few diff fixes rolled up - in case we want to do a 1.3.4 release. 

can I have a review from say: 

@sophistifunk @kzantow @i386 as appropriate (most of the code is from master, but some had to be slightly rewritten if you diff it with master for the apis in use here)

**Comment by michaelneale** [comment]
cc @AlanFoster too - if you could review your change made it safely in here for log improvements (so we can cut a release)

**Comment by AlanFoster** [comment]
@michaelneale Thanks for looking into this; Running this locally everything seems good to me :+1:

File: blueocean-pipeline-api-impl/src/test/java/io/jenkins/blueocean/rest/impl/pipeline/AbstractRunImplTest.java [hunk 1]
```diff
@@ -148,7 +148,7 @@ public void testArtifactZipFileLink() throws Exception {
 
     @Test(timeout = 20000)
     @Issue("JENKINS-44736")
-    public void earlyUnstableStatusShouldReportPunStateAsRunningAndResultAsUnknown() throws Exception {
```
Comment (sophistifunk) [inline blueocean-pipeline-api-impl/src/test/java/io/jenkins/blueocean/rest/impl/pipeline/AbstractRunImplTest.java]: TODO: Dream up a flimsy reason we should have something called a `PunState`

**Comment by sophistifunk** [comment]
Seems pretty straightforward to me 🐝

**Comment by michaelneale** [comment]
@vivek  if ok with you - can we merge this and cut a 1.3.4 ?

**Comment by vivek** [comment]
LGTM 🐝

**Pull request merged**